lms/static/js/views/fields.js

```diff
@@ -495,6 +495,203 @@
+                    url: this.options.profileImageRemoveUrl,
+                    success: function (data, status, xhr) {
+                        self.successHandler();
+                    },
+                    error: function (xhr, status, error) {
+                       self.failureHandler();
+                    }
+                });
+            },
+
+            uploadButtonTitle: function () {
+                if (this.model.has_profile_image()) {
+                    return _.result(this, 'titleEdit')
+                } else {
+                    return _.result(this, 'titleAdd')
+                }
+            },
+
+            removeButtonTitle: function () {
+                return this.titleRemove;
+            },
+
+            successHandler: function (e, data) {
+                var self = this;
+                // Update model to get the latest urls of profile image.
+                this.model.fetch().done(function () {
+                    self.setCurrentStatus('');
+                }).fail(function () {
+                    self.showUserErrorMessage(self.errorMessage);
+                });
+            },
+
+            failureHandler: function (e, data) {
+                this.setCurrentStatus('');
+                 if (_.contains([400, 404], data.jqXHR.status)) {
+                    try {
+                        var errors = JSON.parse(data.jqXHR.responseText);
+                        this.showUserErrorMessage(errors.user_message);
+                    } catch (error) {
+                        this.showUserErrorMessage(this.errorMessage);
+                    }
+                } else {
+                    this.showUserErrorMessage(this.errorMessage);
+                }
+                this.render();
+            },
+
+            fileUploadHandler: function (e, data) {
+                if (this.validateImageSize(data.files[0].size)) {
+                    data.formData = {file: data.files[0]};
+                    this.hideUserErrorMessage();
+                    this.setCurrentStatus('upload');
+                    this.showUploadInProgressMessage();
+                    data.submit();
+                }
+            },
+
+            validateImageSize: function (imageBytes) {
+                if (imageBytes < this.options.imageMinBytes) {
+                    this.showUserErrorMessage(gettext("Minimum file size not met."));
+                    return false;
+                } else if (imageBytes > this.options.imageMaxBytes) {
+                    this.showUserErrorMessage(gettext("Maximum file size exceeded."));
+                    return false;
+                }
+                return true;
+            },
+
+            showUploadInProgressMessage: function () {
+                this.$('.upload-button-wrapper').css('opacity', 1);
+                this.$('.upload-button-icon').html(this.iconProgress);
+                this.$('.upload-button-title').html(gettext("Uploading..."));
```

Comment:
You can make the text a property `titleUploading`. And ellipses are not recommended for accessibility reasons.